### PR TITLE
Fix cell testing issues

### PR DESCRIPTION
**Pull request by azinneera**

### CODE REVIEW

File: components/cli/pkg/commands/project/init.go [hunk 1]
```diff
@@ -263,7 +268,11 @@ func writeCellTemplate(writer io.Writer, cellTemplate string) error {
 // createBalFile creates a bal file at the current location.
 func createTestBalFile(balFilePath string) (*os.File, *string, error) {
 	projectTestDir := filepath.Join(filepath.Dir(balFilePath), "tests")
-	err := util.CreateDir(projectTestDir)
+	isExistsTests, err := util.FileExists(projectTestDir)
```
Comment (isurulucky) [inline components/cli/pkg/commands/project/init.go]: return error

File: components/cli/pkg/commands/image/test.go [hunk 2]
```diff
@@ -346,13 +260,98 @@ func startTestCellInstance(cli cli.Cli, imageDir string, instanceName string, ru
...
+		if !isTestDirExists && !containsTests {
+			return fmt.Errorf("no tests found in the cell image %v", imageTag)
+		}
+		err = util.RemoveDir(testsRoot)
+		if err != nil {
+			return fmt.Errorf("error occurred while deleting filepath, %s", err)
+		}
+
+		// Create Ballerina project
+		isBallerinaProject, err = util.FileExists(balTomlPath)
```
Comment (isurulucky) [inline components/cli/pkg/commands/image/test.go]: return the error

File: components/cli/pkg/commands/image/test.go [hunk 3]
```diff
@@ -346,13 +260,98 @@ func startTestCellInstance(cli cli.Cli, imageDir string, instanceName string, ru
...
+
+		if instanceName != "" {
+			balModule = filepath.Join(testsRoot, constants.ZipBallerinaSource, instanceName)
+		} else {
+			balModule = filepath.Join(testsRoot, constants.ZipBallerinaSource, constants.TempTestModule)
+		}
+		isTestDirExists, _ := util.FileExists(testsPath)
+		var isBallerinaProject bool
+
+		read, err := ioutil.ReadFile(balFilePath)
```
Comment (isurulucky) [inline components/cli/pkg/commands/image/test.go]: return the error.

File: components/cli/pkg/commands/image/test.go [hunk 4]
```diff
@@ -346,13 +260,98 @@ func startTestCellInstance(cli cli.Cli, imageDir string, instanceName string, ru
...
+			return fmt.Errorf("error occurred while creating %s, %v", filepath.Join(testsRoot, "logs"), err)
+		}
+
+		// Change working dir to Bal project
+		err = os.Chdir(testsRoot)
+		if err != nil {
+			return fmt.Errorf("error occurred while changing working directory, %v", err)
+		}
+
+		isExistsSourceBal, err := util.FileExists(filepath.Join(balModule, filepath.Base(balFilePath)))
```
Comment (isurulucky) [inline components/cli/pkg/commands/image/test.go]: return error

File: components/cli/pkg/commands/image/test.go (hunk 3 above)
Comment (azinneera) [reply components/cli/pkg/commands/image/test.go]: Fixed

File: components/cli/pkg/commands/image/test.go (hunk 2 above)
Comment (azinneera) [reply components/cli/pkg/commands/image/test.go]: Fixed

File: components/cli/pkg/commands/image/test.go (hunk 4 above)
Comment (azinneera) [reply components/cli/pkg/commands/image/test.go]: Fixed

File: components/cli/pkg/commands/project/init.go (hunk 1 above)
Comment (azinneera) [reply components/cli/pkg/commands/project/init.go]: Fixed

**Pull request merged**